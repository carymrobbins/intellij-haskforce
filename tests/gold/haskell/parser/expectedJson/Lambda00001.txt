[{"Module":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":4,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1}]},{"ModuleHead":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":16}]},{"ModuleName":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":8},"srcInfoPoints":[]},"Lambda1"]},null,null]},[],[],[{"PatBind":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":31,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},"srcInfoPoints":[]},{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":5,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":5,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},"srcInfoPoints":[]},"zip'"]}]},null,{"UnGuardedRhs":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":31,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":6},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":6}]},{"App":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":31,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":8},"srcInfoPoints":[]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":8},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":8},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":8},"srcInfoPoints":[]},"zipWith"]}]}]},{"Paren":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":31,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":16},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":16},{"srcSpanFilename":"A.hs","srcSpanEndColumn":31,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":30}]},{"Lambda":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":17},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":17},{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":22}]},[{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":18},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":18},"srcInfoPoints":[]},"a"]}]},{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":20},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":20},"srcInfoPoints":[]},"b"]}]}],{"Tuple":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":25},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":25},{"srcSpanFilename":"A.hs","srcSpanEndColumn":28,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":27},{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":29}]},"Boxed",[{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":26},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":26},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":26},"srcInfoPoints":[]},"a"]}]}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":29,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":28},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":29,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":28},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":29,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":28},"srcInfoPoints":[]},"b"]}]}]}]]}]}]}]}]},null]}]]},[]]