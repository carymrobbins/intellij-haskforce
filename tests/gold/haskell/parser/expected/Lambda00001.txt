Haskell File
  MODULEDECL
    PsiElement(HaskellTokenType.module)('module')
    QCONID
      CONID
        PsiElement(HaskellTokenType.conidRegexp)('Lambda1')
    PsiElement(HaskellTokenType.where)('where')
  BODY
    FUNORPATDECL
      PAT
        VARID
          PsiElement(HaskellTokenType.varidRegexp)('zip'')
      RHS
        PsiElement(HaskellTokenType.=)('=')
        EXP
          QVAR
            QVARID
              VARID
                PsiElement(HaskellTokenType.varidRegexp)('zipWith')
          PsiElement(HaskellTokenType.()('(')
          EXP
            PsiElement(HaskellTokenType.\)('\')
            VARID
              PsiElement(HaskellTokenType.varidRegexp)('a')
            VARID
              PsiElement(HaskellTokenType.varidRegexp)('b')
            PsiElement(HaskellTokenType.->)('->')
            EXP
              PsiElement(HaskellTokenType.()('(')
              EXP
                QVAR
                  QVARID
                    VARID
                      PsiElement(HaskellTokenType.varidRegexp)('a')
              PsiElement(HaskellTokenType.,)(',')
              EXP
                QVAR
                  QVARID
                    VARID
                      PsiElement(HaskellTokenType.varidRegexp)('b')
              PsiElement(HaskellTokenType.))(')')
          PsiElement(HaskellTokenType.))(')')